### PR TITLE
Add left/right positioning, update tooltip style, update positioning

### PR DESCRIPTION
**Pull request by moriko**
cc @nitruxa, @nerijussavickas, @hellatan, @jrdrg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ const Tooltip = require('dibs-react-tooltip');
 
 render() {
 	return (
-		<Tooltip isVisible={true} rootPosition={{x: 50, y: 50}}>
+		<Tooltip isVisible={true}>
```
Comment (hellatan) [inline README.md]: I don't think there's a linter in this project. You don't need to assign a value to isVisible if you are defaulting it to true. Not sure if you'd see an error in the browser console though

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ render() {
 Props:
 ```
 // Determines the initial position of the tooltip relative to rootPosition
-tooltipDirection: 		React.PropTypes.oneOf(['top', 'bottom']),	// TODO: left and right
+tooltipDirection: 		React.PropTypes.oneOf(['top', 'bottom', 'left', 'right']),	// TODO: left and right
```
Comment (hellatan) [inline README.md]: You can remove the "todo"

Comment (moriko) [inline README.md]: Probably update the "todo" as still need to think of what to do with the tooltip when it's set "left" and there's no space on the left. Should it be shown on the right? But if there's not space on the right then bottom?

File: README.md (hunk 1 above)
Comment (moriko) [inline README.md]: There's no default value for `isVisible`.

File: Tooltip.jsx [hunk 3]
```diff
@@ -81,6 +57,41 @@ class Tooltip extends React.Component {
...
     }
 
+    _getInitialTooltipPosition(elementRect, tooltipDirection, rootPosition) {
+        const { hasTriangle, triangleSize } = this.props;
+        const triangleSpacing = hasTriangle ? triangleSize : (triangleSize / 2);
+        const left = -1 * (elementRect.width / 2) + (rootPosition.width / 2);
+        const top = -1 * (elementRect.height / 2) + (rootPosition.height / 2);
+        const triangleLeft = (elementRect.width / 2) - triangleSpacing;
+        const triangleTop = (elementRect.height / 2) - (triangleSize / 2);
+        const spacing = triangleSpacing + 3;
```
Comment (hellatan) [inline Tooltip.jsx]: you can DRY this up a bit like this:

```js
function getElementDims(el) {
    return function (type) {
        if (el.width && type === 'width') {
            return el.width / 2;
        }
        if (el.height && type === 'height') {
            return el.height / 2;
        }
    }
}

const elDims = getElementDims(elementRect);
const rootDims = getElementDims(rootPosition);

const elHalfWidth = elDims('width');
const elHalfHeight = elDims('height');
const rootHalfWidth = rootDims('width');
const rootHalfHeight = rootDims('height');

const left = -1 * elHalfWidth + rootHalfWidth;
const top = -1 * elHalfHeight + rootHalfHeight;
const triangleLeft = elHalfWidth - triangleSpacing;
const triangleTop = elHalfHeight - (triangleSize / 2);
const spacing = triangleSpacing + 3;
```

I think having the descriptive variable names also makes it easier to digest what type of values you are using. You can probably take it a little further with a functional programming approach, but that's a little too much for me to think about right now 😁

File: Tooltip.jsx [hunk 4]
```diff
@@ -61,7 +36,8 @@ class Tooltip extends React.Component {
 
     componentWillReceiveProps(nextProps) {
         this.setState({
-            tooltipDirection: nextProps.tooltipDirection || this.state.tooltipDirection
+            tooltipDirection: nextProps.tooltipDirection || this.state.tooltipDirection,
+            isVisible: nextProps.isVisible
```
Comment (nitruxa) [inline Tooltip.jsx]: I think we can run into a problem here.
`_onCloseClick` function sets state `isVisible` to `false`, but when root component re-renders, it still has `tooltipIsVisible` (or whatever) prop as `true`, cause it was once opened, and will pass it here, so `isVisible` state will be updated to `true` again, even it has to be closed.

Comment (moriko) [inline Tooltip.jsx]: Totally correct, thanks for catching this :+1:

**Comment by hellatan** [comment]
@moriko is there a ticket for this?

**Comment by gedas-monginas** [comment]
@hellatan this is a public repo :) I don't think we need to add ticket number in these commits

**Comment by hellatan** [comment]
ooh didn't realize that.

**Comment by nitruxa** [comment]
@gedas-monginas merge! 🚢

**Comment by gedas-monginas** [comment]
:fireworks:

**Pull request merged**